### DISCUSSION TITLE
PML_x: Perfectly Matched Layer absorbing boundary condition, using x number of cells

### DISCUSSION THREAD

**Discussion by noor0re** [thliebig/openEMS-Project#25] [Q&A Usage] [answered]
Hello every one. 
could some one explain to me what means "cells" here?

**Comment by 0xCoto** [comment, marked as answer]
Mesh cells (grid lines).

**Comment by biergaizi** [comment]
In electromagnetic simulations, all waves only exist in a small simulation box, so you need to do something when they reach the edge, this is the physical intuition of boundary condition in PDE. The simplest solution is setting the electric or magnet field outside the box to zero, mathematically it's known as the Dirichlet boundary condition, physically it represents the behavior of a Perfect Electric Conductor or a Perfect Magnetic Conductor (short or open circuit in 1D), but they reflect the waves back into the box. It's useful when you're actually simulating a shielded box (it also has the fastest simulation speed), such as a metal enclosure, a ground plane or a waveguide. But for open boundary problems, the reflection must be eliminated, so there must be an Absorbing Boundary Condition to make the waves disappear at the boundary.

The first generation of ABCs are differential-equation based, openEMS's Mur ABC belongs to this category. If used properly, it works, and the overhead is low, but absorption is only ideal when the angle of incidence is normal to the ABC, at other angles it doesn't work well. The second generation of ABCs is the Perfectly Matched Layer (PML).  Instead of a real mathematical boundary condition, PML behaves as an unusual artificial material that can magically absorb almost all the E&M waves perfectly. It's easy to use and works much better, but with a higher computational overhead (especially in openEMS, since its current implementation is not optimal, speed-wise), so a tradeoff between speed vs. ease of use vs. accuracy must be made. Also remember that for all ABCs, including PML, there do exist special cases under which they can fail or even destabilize the simulation.

Back to the question, PML is a material, not a true boundary condition in the mathematical sense, and it occupies physical space in the simulation box. Using PML_8, it means the PML material occupies the last 8 cells at edge of the mesh grid and it's a good default. Depending on the problem, a thicker "foam" (so to speak) may be needed, so the number is adjustable.

**Reply by noor0re** [reply]
Many thanks for the instructions about boundary conditions and their relation with resources needed to solve the equation with numerical methods.

**Reply by thliebig** [reply]
Yes indeed, nice summary. We have to make sure more of this makes it into the documentation...

Some additions/corrections from me.
The Mur ABC are unconditionally stable, its **only** the PML that is potentially unstable, especially if fringe fields are entering them... Thus make sure to keep a distance of any radiating structure of at least lambda/4...

**Reply by biergaizi** [reply]
Thanks for the correction.
